library/network_connections.py

```diff
@@ -686,7 +686,13 @@ def setting_ip_config_get_routes(self, s_ip):
     def connection_ensure_setting(self, connection, setting_type):
         setting = connection.get_setting(setting_type)
         if not setting:
-            setting = setting_type.new()
+            if setting_type is Util.NM().SettingMatch:
+                # https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/4d609dd92f2922e92888e02b5b857e4d7b10a37e
+                # Note that function 'nm_setting_match_new()' was only exposed and usable since libnm version 1.32
+                # The workaround is to always use g_object_new(NM_TYPE_SETTING_MATCH), which works for all libnm versions since 1.14.
+                setting = setting_type()
+            else:
+                setting = setting_type.new()
```

Comment:
```suggestion
            setting = setting_type()
```
If the `.new()` functions are not needed, this seems to be the best solution. @thom311 told me that nmcli is not using the `.new()` functions, too.

Reply:
we already discussed this on this PR, and I suggested to keep using the new function because that is more idomatic.

But not that it makes any difference...

Reply:
back then I understood that it is mostly the same not exactly the same. Using `setting_type()` instead of `setting_type.new()` is more pythonic.

Reply:
Since it did not make any difference functionally, then call `setting_type()` all the time. Updated.